### PR TITLE
Fix links to supported hardware in README.md

### PR DESCRIPTION
**Pull request by pietrodn**
Some of the links to supported hardware boards were redirected, or broken.
I also added some links for the boards without any.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,24 +74,24 @@ The PX4 Dev Team syncs up on a [weekly dev call](https://dev.px4.io/en/contribut
...
   * FMUv4.x
-    * [Pixracer](https://dev.px4.io/en/flight_controller/pixracer.html)
-    * Pixhawk 3 Pro
+    * [Pixracer](https://docs.px4.io/en/flight_controller/pixracer.html)
+    * [Pixhawk 3 Pro](https://docs.px4.io/en/flight_controller/pixhawk3_pro.html
   * FMUv5.x (ARM Cortex M7, future Pixhawk)
-  * STM32F4Discovery (basic support) [Tutorial](https://pixhawk.org/modules/stm32f4discovery)
-  * Gumstix AeroCore (v1 and v2)
+  * [STM32F4Discovery](http://www.st.com/en/evaluation-tools/stm32f4discovery.html) (basic support) [Tutorial](https://pixhawk.org/modules/stm32f4discovery)
+  * [Gumstix AeroCore](https://www.gumstix.com/aerocore-2/) (v1 and v2)
```
Comment (dagar) [inline README.md]: Only v2 is supported actually.

File: README.md [hunk 2]
```diff
@@ -74,24 +74,24 @@ The PX4 Dev Team syncs up on a [weekly dev call](https://dev.px4.io/en/contribut
...
-    * [Pixfalcon](https://dev.px4.io/en/flight_controller/pixfalcon.html)
+    * [Pixhawk](https://docs.px4.io/en/flight_controller/pixhawk.html)
+    * [Pixhawk Mini](https://docs.px4.io/en/flight_controller/pixhawk_mini.html)
+    * [Pixfalcon](https://docs.px4.io/en/flight_controller/pixfalcon.html)
   * FMUv3.x [Pixhawk 2](https://pixhawk.org/modules/pixhawk2)
   * FMUv4.x
-    * [Pixracer](https://dev.px4.io/en/flight_controller/pixracer.html)
-    * Pixhawk 3 Pro
+    * [Pixracer](https://docs.px4.io/en/flight_controller/pixracer.html)
+    * [Pixhawk 3 Pro](https://docs.px4.io/en/flight_controller/pixhawk3_pro.html
```
Comment (hamishwillee) [inline README.md]: Missing end bracket on link

**Comment by hamishwillee** [comment]
Thanks @pietrodn - much appreciated! There are some minor issues with this but it is all much better, so I'll merge and fix those.

**Comment by hamishwillee** [comment]
Thanks @pietrodn - much appreciated! There are some minor issues with this in line.

@dagar This is good. I don't have rights to merge in this repo. There are other issues with this readme outside of this section that I will fix after this is merged.

**Comment by pietrodn** [comment]
I just integrated your corrections into the PR.

**Pull request merged**